sample-extension-plugin/src/test/java/org/opensearch/jobscheduler/sampleextension/bwc/JobSchedulerBackwardsCompatibilityIT.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.jobscheduler.sampleextension.bwc;
+
+import org.junit.Assert;
+import org.opensearch.jobscheduler.sampleextension.SampleExtensionIntegTestCase;
+import org.opensearch.test.rest.OpenSearchRestTestCase;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class JobSchedulerBackwardsCompatibilityIT extends SampleExtensionIntegTestCase {
+    private static final ClusterType CLUSTER_TYPE = ClusterType.parse(System.getProperty("tests.rest.bwcsuite"));
+    private static final String CLUSTER_NAME = System.getProperty("tests.clustername");
+
+    @SuppressWarnings("unchecked")
+    public void testBackwardsCompatibility() throws Exception {
+        String uri = getPluginUri();
+        assert uri != null;
+
+        Map<String, Object> response = getAsMap(uri);
+        Map<String, Map<String, Object>> responseMap = (Map<String, Map<String, Object>>) response.get("nodes");
+
+        for (Map<String, Object> respValues: responseMap.values()) {
+            List<Map<String, Object>> plugins = (List<Map<String, Object>>) respValues.get("plugins");
+            List<String> pluginNames = plugins.stream().map(plugin -> plugin.get("name").toString()).collect(Collectors.toList());
+
+            switch (CLUSTER_TYPE) {
+                case OLD:
+                    Assert.assertTrue(pluginNames.contains("opendistro-job-scheduler"));
+                    break;
+                case MIXED:
+                    Assert.assertTrue(pluginNames.contains("opensearch-job-scheduler"));
+                    break;
+                case UPGRADED:
+                    Assert.assertTrue(pluginNames.contains("opensearch-job-scheduler"));
+                    createBasicWatcherJob();
```

Comment:
Could you add a comment about the strategy of the BWC test? It would be difficult for someone with no context to realize why you are using the hardcoded string with no explanation. I would also like to see some more comments throughout, I always find the BWC tests a bit difficult to read.

Reply:
+1
I am wondering why only do the `createBasicWatcherJob` in UPGRADED but not OLD and MIXED.

Reply:
From the gradle file bwc tests setup, it looks to me that the rolling upgrade bwc tests are using the same cluster `testClusters."${baseName}0`, so prob you only need to create job at the OLD cluster, and only assert it's running at MIXED and UPGRADED cluster.

Reply:
hi @bowenlan-amzn , we cannot create a job in old cluster as sample-extension plugin is not loaded as old zip file is not available